src/web/components/StickyBottomBanner/BrazeBanner.tsx

```diff
@@ -0,0 +1,135 @@
+`;
+
+export const hasRequiredConsents = (): Promise<boolean> =>
+    new Promise((resolve) => {
+        onConsentChange(({ tcfv2, ccpa }) => {
+            const consentGivenUnderCcpa = ccpa && !ccpa.doNotSell;
+            const consentGivenUnderTcfv2 =
+                tcfv2 && tcfv2.vendorConsents[brazeVendorId];
+
+            resolve(!!(consentGivenUnderCcpa || consentGivenUnderTcfv2));
+        });
+    });
+
+export const canShow = async (
+    asyncBrazeUuid: Promise<null | string>,
+    isDigitalSubscriber: undefined | boolean,
+): Promise<CanShowResult> =>
+    new Promise((resolve) => {
+        const { brazeSwitch } = window.guardian.config.switches;
+        const apiKey = window.guardian.config.page.brazeApiKey;
+
+        if (!(brazeSwitch && apiKey)) {
+            resolve({ result: false });
+            return;
+        }
+
+        if (isDigitalSubscriber) {
+            resolve({ result: false });
+            return;
+        }
+
+        Promise.all([asyncBrazeUuid, hasRequiredConsents()]).then(
+            ([brazeUuid, hasGivenConsent]) => {
+                if (!(brazeUuid && hasGivenConsent)) {
+                    resolve({ result: false });
+                } else {
+                    import(
+                        /* webpackChunkName: "braze-web-sdk-core" */ '@braze/web-sdk-core'
+                    )
+                        .then(({ default: appboy }) => {
+                            appboy.initialize(apiKey, {
+                                noCookies: true,
+                                baseUrl: 'https://sdk.fra-01.braze.eu/api/v3',
+                                sessionTimeoutInSeconds: 1,
+                                minimumIntervalBetweenTriggerActionsInSeconds: 0,
+                            });
+
+                            appboy.subscribeToInAppMessage((message) => {
+                                const meta = (message as any).extras;
+
+                                if (meta) {
+                                    resolve({ result: true, meta });
+                                } else {
+                                    resolve({ result: false });
+                                }
+                            });
+
+                            appboy.changeUser(brazeUuid);
+                            appboy.openSession();
+                        })
+                        .catch(() => {
+                            resolve({ result: false });
+                        });
+                }
+            },
+        );
+    });
+
+export const BrazeBanner = ({ meta }: Props) => {
+    const [ExampleComponent, setExampleComponent] = useState<
+        React.FC<BrazeBannerProps>
+    >();
+
+    useEffect(() => {
+        if (meta && meta['test-key']) {
+            // TODO: unify the way we handle sharing these deps (this is
+            // duplicated in SlotBodyEnd). Probably via the automat client
+            // library.
+            window.guardian.automat = {
```

Comment:
Let's use `automat-modules` for this, but can be a separate PR (which I'm happy to raise for here and slot body end too).

Reply:
Yep good point. Let's add that in a separate PR.